File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/documents/DocumentInfoRepositoryIntegrationTest.java
package io.zenwave360.example.clinicaltool.modules.documents;

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest;
import io.zenwave360.example.clinicaltool.modules.documents.domain.*;
import jakarta.persistence.EntityManager;
import java.time.*;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class DocumentInfoRepositoryIntegrationTest extends BaseRepositoryIntegrationTest {

    @Autowired
    EntityManager entityManager;

    @Autowired
    DocumentInfoRepository documentInfoRepository;

    @Test
    void findAllTest() {
        var results = documentInfoRepository.findAll();
        Assertions.assertFalse(results.isEmpty());
    }

    @Test
    void findByIdTest() {
        var id = 1L;
        var documentInfo = documentInfoRepository.findById(id).orElseThrow();
        Assertions.assertNotNull(documentInfo.getId());
        Assertions.assertNotNull(documentInfo.getVersion());
    }

    @Test
    void saveTest() {
        DocumentInfo documentInfo = new DocumentInfo();
        documentInfo.setUuid("");
        documentInfo.setFileName("");
        documentInfo.setDocumentType("");
        documentInfo.setContentType("");
        documentInfo.setTags(List.of(""));

        // OneToOne documentData owner: true
        var documentDataId = 1L;
        var documentData = new DocumentData();
        documentData.setData("test document content".getBytes());
        documentInfo.setDocumentData(documentData);

        // Persist aggregate root
        var created = documentInfoRepository.save(documentInfo);

        // reloading to get relationships persisted by id
        entityManager.flush();
        entityManager.refresh(created);
        Assertions.assertNotNull(created.getId());
        Assertions.assertNotNull(created.getVersion());

        Assertions.assertNotNull(documentInfo.getDocumentData().getId() != null);
    }

    @Test
    void updateTest() {
        var id = 1L;
        var documentInfo = documentInfoRepository.findById(id).orElseThrow();
        documentInfo.setUuid("");
        documentInfo.setFileName("");
        documentInfo.setDocumentType("");
        documentInfo.setContentType("");
        documentInfo.setTags(List.of(""));

        documentInfo = documentInfoRepository.save(documentInfo);
        Assertions.assertEquals("", documentInfo.getUuid());
        Assertions.assertEquals("", documentInfo.getFileName());
        Assertions.assertEquals("", documentInfo.getDocumentType());
        Assertions.assertEquals("", documentInfo.getContentType());
        Assertions.assertEquals(List.of(""), documentInfo.getTags());
    }

    @Test
    void deleteTest() {
        var id = 1L;
        documentInfoRepository.deleteById(id);
        var notFound = documentInfoRepository.findById(id);
        Assertions.assertFalse(notFound.isPresent());
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/documents/DocumentInfoRepositoryIntegrationTest.kt
package io.zenwave360.example.clinicaltool.modules.documents

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest
import io.zenwave360.example.clinicaltool.modules.documents.domain.*
import jakarta.persistence.EntityManager
import java.time.*
import java.util.List
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.repository.findByIdOrNull

class DocumentInfoRepositoryIntegrationTest : BaseRepositoryIntegrationTest() {

    @Autowired lateinit var entityManager: EntityManager

    @Autowired lateinit var documentInfoRepository: DocumentInfoRepository

    @Test
    fun findAllTest() {
        val results = documentInfoRepository.findAll()
        Assertions.assertFalse(results.isEmpty())
    }

    @Test
    fun findByIdTest() {
        val id = 1L
        val documentInfo =
            documentInfoRepository.findByIdOrNull(id) ?: throw NoSuchElementException(" not found with id: $id")
        Assertions.assertNotNull(documentInfo.id)
        Assertions.assertNotNull(documentInfo.version)
    }

    @Test
    fun saveTest() {
        val documentInfo = DocumentInfo()
        documentInfo.uuid = ""
        documentInfo.fileName = ""
        documentInfo.documentType = ""
        documentInfo.contentType = ""
        documentInfo.tags = mutableListOf("")

        // OneToOne documentData owner: true
        val documentDataId = 1L
        val documentData = DocumentData()
        documentData.data = "test document content".toByteArray()
        documentInfo.documentData = documentData

        // Persist aggregate root
        val created = documentInfoRepository.save(documentInfo)

        // reloading to get relationships persisted by id
        entityManager.flush()
        entityManager.refresh(created)
        Assertions.assertNotNull(created.id)
        Assertions.assertNotNull(created.version)

        Assertions.assertNotNull(documentInfo.documentData?.id != null)
    }

    @Test
    fun updateTest() {
        val id = 1L
        val documentInfo =
            documentInfoRepository.findByIdOrNull(id) ?: throw NoSuchElementException(" not found with id: $id")
        documentInfo.uuid = ""
        documentInfo.fileName = ""
        documentInfo.documentType = ""
        documentInfo.contentType = ""
        documentInfo.tags = mutableListOf("")

        val updated = documentInfoRepository.save(documentInfo)
        Assertions.assertEquals("", updated.uuid)
        Assertions.assertEquals("", updated.fileName)
        Assertions.assertEquals("", updated.documentType)
        Assertions.assertEquals("", updated.contentType)
        Assertions.assertEquals(mutableListOf(""), updated.tags)
    }

    @Test
    fun deleteTest() {
        val id = 1L
        documentInfoRepository.deleteById(id)
        val notFound = documentInfoRepository.findByIdOrNull(id)
        Assertions.assertNull(notFound)
    }
}
